frontend/src/pages/users/UserAdd.tsx

```diff
@@ -0,0 +1,275 @@
+import { FormEvent, useEffect, useRef, useState } from "react";
+import { logger } from "../../util/Logger";
+import { Toast } from "primereact/toast";
+import BackButton from "../../components/buttons/BackButton";
+import { useNavigate, useParams } from "react-router-dom";
+import { InputText } from "primereact/inputtext";
+import { Password } from "primereact/password";
+import { InputSwitch } from "primereact/inputswitch";
+import {
+  passwordValueCheck,
+  pwCheckFooter,
+  pwCheckHeader,
+} from "../../util/PWValueCheck";
+import { showFailure, showSuccess } from "../../components/Toast";
+import ConfirmPopup from "../../components/popups/ConfirmPopup";
+import DeletePopup from "../../components/popups/DeletePopup";
+import DeleteButton from "../../components/buttons/DeleteButton";
+import { USER_API } from "../../apis/users/UserAPI";
+
+export default function VendorAdd() {
+  // From URL
+  const { id } = useParams();
+  const isUserAddPage = id === undefined;
+  const [isModifiable] = useState<boolean>(true);
+
+  const [userName, setUserName] = useState<string>("");
+  const [password1, setPassword1] = useState<string>("");
+  const [password2, setPassword2] = useState<string>("");
+  const [isAdmin, setIsAdmin] = useState<boolean>(false);
+
+  const [isConfirmationPopupVisible, setIsConfirmationPopupVisible] =
+    useState<boolean>(false);
+  const [deletePopupVisible, setDeletePopupVisible] = useState<boolean>(false); // Whether the delete popup is visible
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
If these variables are unused you should just get rid of them

Reply:
This variable is needed for future use. That's why I left it in.